flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/overlays/FlinkDistributionOverlay.java

```diff
@@ -75,7 +75,12 @@ public void configure(ContainerSpecification container) throws IOException {
 		addPathRecursively(flinkBinPath, TARGET_ROOT, container);
 		addPathRecursively(flinkConfPath, TARGET_ROOT, container);
 		addPathRecursively(flinkLibPath, TARGET_ROOT, container);
-		addPathRecursively(flinkPluginsPath, TARGET_ROOT, container);
+		if (flinkPluginsPath.isDirectory()) {
```

Comment:
Why is it obligatory to set `ENV_FLINK_PLUGINS_DIR` for mesos but not for yarn? If I do not miss something `FlinkDistributionOverlay.Builder.fromEnvironment.getObligatoryFileFromEnvironment(ENV_FLINK_PLUGINS_DIR)` will fail if `ENV_FLINK_PLUGINS_DIR` is not set.

Reply:
Hmm, good question. I was just repeating the same pattern as with `ENV_FLINK_LIB_DIR`. So asking the correct question: why is it obligatory to set `ENV_FLINK_LIB_DIR` for mesos but not for yarn? I have no idea. 

If you want, we can create a ticket to unify those two things - which first step would be to find out which way they should be unified.

Reply:
True, we can create a separate ticket. The reason, I brought up it here, is that the PR basically addresses the problem (or at least some of them) that the previous deployments (w/o plugins) are not broken after plugins have been introduced. I guess the previous mesos deployments (same as yarn) do not have `ENV_FLINK_PLUGINS_DIR` and this case should be probably also tolerated, can be a separate issue though if you think it needs more discussion.

Reply:
As discussed this concern might be not an issue currently because as long as the user runs our `flink-dist` scripts to submit mesos jobs, `config.sh` has a default value for `FLINK_PLUGINS_DIR` which might not exist though but should work overall with this fix. The plugins shipping can only break deployments with custom scripts, same as with `lib` and other directories.